assembly-ide/src/assembly/bin/che.sh

```diff
@@ -10,20 +10,451 @@
+     start                   Starts Che application server in new console
+     stop                    Stops Che application server"
+
+  # Command line parameters
+  USE_DOCKER=false
+  CHE_DOCKER_TAG=latest
+  CHE_PORT=8080
+  USE_HELP=false
+  CHE_SERVER_ACTION=run
+  USE_DEBUG=false
+
+  # Sets value of operating system
+  WIN=false
+  MAC=false
+  LINUX=false
+}
+
+function usage {
+  echo "$USAGE"
+}
+
+function error_exit {
+  echo
+  echo "$1"
+  echo "$WRONG $CHE_VARIABLES $USAGE"
+  JUMP_TO_END=true
+}
+
+function parse_command_line {
+
+  for command_line_option in "$@"
+  do
+  case $command_line_option in
+    -i|--image)
+      USE_DOCKER=true
+    ;;
+    -i:*|--image:*)
+      USE_DOCKER=true
+      if [ "${command_line_option#*:}" != "" ]; then
+        CHE_DOCKER_TAG="${command_line_option#*:}"
+      fi
+    ;;
+    -p:*|--port:*)
+      CHE_PORT="${command_line_option#*:}"
+    ;;
+    -h|--help)
+      USE_HELP=true
+      usage
+      return
+    ;;
+    -d|--debug)
+      USE_DEBUG=true
+    ;;    
+    start|stop|run)
+      CHE_SERVER_ACTION=${command_line_option}
+    ;;
+    *)
+      # unknown option
+      error_exit "!!! You passed an unknown command line option."
+    ;;
+  esac
+  done
+
+  if $USE_DEBUG; then
+    echo "USE_DOCKER: ${USE_DOCKER}"
+    echo "CHE_DOCKER_TAG: ${CHE_DOCKER_TAG}"
+    echo "CHE_PORT: ${CHE_PORT}"
+    echo "USE_HELP: ${USE_HELP}"
+    echo "CHE_SERVER_ACTION: ${CHE_SERVER_ACTION}"
+    echo "USE_DEBUG: ${USE_DEBUG}"
+  fi
+}
+
+function determine_os {
+  # Set OS.  Mac & Windows require VirtualBox and docker-machine.
+
+  if [[ "${OSTYPE}" == "linux-gnu" ]]; then
+    # Linux
+    LINUX=true
+  elif [[ "${OSTYPE}" == "darwin"* ]]; then
+    # Mac OSX
+    MAC=true
+  elif [[ "${OSTYPE}" == "cygwin" ]]; then
+    # POSIX compatibility layer and Linux environment emulation for Windows
+    WIN=true
+  elif [[ "${OSTYPE}" == "msys" ]]; then
+    # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
+    WIN=true
+  elif [[ "${OSTYPE}" == "win32" ]]; then
+    # I'm not sure this can happen.
+    WIN=true
+  elif [[ "${OSTYPE}" == "freebsd"* ]]; then
+    # FreeBSD
+    LINUX=true
+  else
+    error_exit "We could not detect your operating system. Che is unlikely to work properly."
+  fi
+
+}
+
+function set_environment_variables {
+  ### Set the value of derived environment variables.
+  ### Use values set by user, unless they are broken, then fix them
+  # The base directory of Che
+  if [ -z "${CHE_HOME}" ] || [ ! -f "${CHE_HOME}" ]; then
+    export CHE_HOME="$(dirname "$(cd $(dirname ${0}) && pwd -P)")"
+  fi
+
+  # Convert Tomcat environment variables to POSIX format.
+  if [[ "${JAVA_HOME}" == *":"* ]]
+  then 
+    JAVA_HOME=$(echo /$JAVA_HOME | sed  's|\\|/|g' | sed 's|:||g')
+  fi
+
+  if [[ "${CHE_HOME}" == *":"* ]]
+  then 
+    CHE_HOME=$(echo /$CHE_HOME | sed  's|\\|/|g' | sed 's|:||g')
```

Comment:
Yes, this would be nice.  It is giving me some challenges because the variable changes.